lisa/env.py

```diff
@@ -437,22 +472,31 @@ def get_res_dir(self, name=None, append_time=True, symlink=True):
         """
         logger = self.get_logger()
 
+        if isinstance(self._res_dir, ArtifactPath):
```

Comment:
There already is a use of the pathlib API in `get_res_dir()` that slipped my attention, but I would like to refrain from using it (it's probably too late for that in exekall/bisector but it can still be done for LISA).

A large amount of bugs and issues we've had with the next branch was due to this API breaking compatibility between Python 3.5/3.6/3.7, and I'd rather spend my time on other things than to constantly fight those and hack up workarounds.

We might envision embracing it once the API *actually* stabilizes and we bump up the Python 3 version requirement, but until now I believe we'd be better off without it.